### PR TITLE
Split SmartOS related grain off into a seperate module

### PR DESCRIPTION
**Pull request by sjorge**
### What does this PR do?
Moves the SmartOS related grains (both computenode and native zones) into a separate module (salt.grains.smartos)

In addition to this some of the costly execs were reduced from 4 to 2, making those grains a bit faster to be populated. Grain info about the type of vms' is now also included.

For zone an extra grain was also added (```pkgin_repositories```). Originally I wanted to fix the pkgsrcpath grain to list all repositories but that grain is based on an environment value and that one does indeed only hold the main repesitory.

### What issues does this PR fix or reference?
N/A

### Previous Behavior
Grains were in salt.grains.core

### New Behavior
Grains are in salt.grains.smartos, extra grain ```pkgin_repositories``` was added, small performance improvement.

### Tests written?
No

### Commits signed with GPG?
No

### CODE REVIEW

File: salt/grains/smartos.py [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+
+    # sysinfo derived smbios grains
+    grains['manufacturer'] = sysinfo['Manufacturer']
+    grains['productname'] = sysinfo['Product']
+    grains['uuid'] = sysinfo['UUID']
+
+    return grains
+
+
+def _smartos_zone_data():
```
Comment (cachedout) [inline salt/grains/smartos.py]: Can we split this out into separate functions? The complexity of this function is quite high.

**Review by cachedout — CHANGES_REQUESTED** [review]

File: salt/grains/smartos.py (hunk 1 above)
Comment (sjorge) [reply salt/grains/smartos.py]: How about splitting this in 3?
- _smartos_zone_pkgsrc_data
- _smartos_zone_pkgin_data
- _smartos_zone_image_data

?

Comment (cachedout) [reply salt/grains/smartos.py]: Perfect!

Comment (sjorge) [reply salt/grains/smartos.py]: @cachedout done

**Comment by sjorge** [comment]
Hmm a whole bunch of test failures, all with:
```u'The salt master could not be contacted. Is master running?'```

Can someone give jenkins another run please?

**Review by cachedout — APPROVED** [review]

**Pull request merged**